server/etcdserver/txn/txn.go

```diff
@@ -37,55 +37,36 @@ func Put(ctx context.Context, lg *zap.Logger, lessor lease.Lessor, kv mvcc.KV, p
 		traceutil.Field{Key: "key", Value: string(p.Key)},
 		traceutil.Field{Key: "req_size", Value: p.Size()},
 	)
-	leaseID := lease.LeaseID(p.Lease)
```

Comment:
This Put is called in the applier code path. Invoking kv.Write to acquire the lock before checking the leasor does not incur significant overhead.